### PR TITLE
Add indexing

### PR DESCRIPTION
**Pull request by cdebourcy**
- added script to automatically generate STAR and bowtie2 indexes from fasta file
- allow location of desired STAR and bowtie2 indexes to be specified as environment variable when kicking off pipeline

### CODE REVIEW

File: aws_batch/indexing.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+import os
+import subprocess
+
+INPUT_FASTA_S3 = 's3://czbiohub-infectious-disease/references/mosquitos/mosquito_genomes2.fa'
+OUTPUT_PATH_S3 = 's3://czbiohub-infectious-disease/references/mosquitos'
+
+ROOT_DIR = '/mnt'
+DEST_DIR = ROOT_DIR + '/idseq/indexes' # generated indexes go here
```
Comment (yunfangjuan) [inline aws_batch/indexing.py]: indices?

File: aws_batch/indexing.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+def make_star_index(fasta_file, result_dir, scratch_dir, output_path_s3, lazy_run):
+    if lazy_run:
+        output = os.path.join(result_dir, STAR_INDEX_OUT)
+        if os.path.isfile(output):
+            return 1
+    star_genome_dir_name = STAR_INDEX_OUT.split('.')[0]
+    star_command_params = ['cd', scratch_dir, ';',
+                           'mkdir', star_genome_dir_name, ';',
+                           STAR, '--runThreadN', '4',
```
Comment (yunfangjuan) [inline aws_batch/indexing.py]: import multiprocessing
4 => 
str(multiprocessing.cpu_count())

File: aws_batch/indexing.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    execute_command(" ".join(bowtie2_command_params))
+    print "finished making bowtie2 index"
+    # archive and compress
+    execute_command("tar czvf %s/%s -C %s %s" % (result_dir, BOWTIE2_INDEX_OUT, scratch_dir, bowtie2_genome_dir_name))
+    # copy to S3
+    execute_command("aws s3 cp %s/%s %s/;" % (result_dir, BOWTIE2_INDEX_OUT, output_path_s3))
+    # cleanup
+    execute_command("cd %s; rm -rf *" % scratch_dir)
+
+def make_indexes(input_fasta_s3, output_path_s3, lazy_run = True):
```
Comment (yunfangjuan) [inline aws_batch/indexing.py]: indices?

**Comment by yunfangjuan** [comment]
lgtm with a few minor comments.

File: aws_batch/indexing.py (hunk 1 above)
Comment (cdebourcy) [reply aws_batch/indexing.py]: In this context people seem to say 'indexes' more often (e.g. STAR manual)

**Comment by yunfangjuan** [comment]
ok then

File: aws_batch/indexing.py (hunk 1 above)
Comment (yunfangjuan) [reply aws_batch/indexing.py]: ok then!

**Pull request merged**